src/bindgen.rs

```diff
@@ -1,16 +1,40 @@
 use emoji;
 use error::Error;
 use progressbar::Step;
-use std::process::Command;
+use std::{fs, process::Command};
 use PBAR;
 
-pub fn cargo_install_wasm_bindgen(step: &Step) -> Result<(), Error> {
+static LOCAL_BINDGEN_PATH: &str = "bin/wasm-bindgen";
+
+pub fn wasm_bindgen_locally_installed(_path: &str) -> bool {
+    fs::metadata(LOCAL_BINDGEN_PATH).is_ok()
+}
+
+pub fn wasm_bindgen_version_check(dep_version: &str) -> Result<bool, Error> {
+    let output = Command::new(LOCAL_BINDGEN_PATH).arg("--version").output()?;
+    if output.status.success() {
+        let stdout = String::from_utf8_lossy(&output.stdout);
+        let installed_version = stdout.trim();
+        Ok(dep_version == installed_version)
+    } else {
+        let message = "Could not find version of local wasm-bindgen".to_string();
+        let stderr = String::from_utf8_lossy(&output.stderr).to_string();
+        let e = Error::Cli { message, stderr };
+        Err(e)
+    }
```

Comment:
Not a giant fan of this :point_up: Creating the `Error:Cli` manually, rather than through the `cli(..)` function seems a little hacky, but the other approach caused a return type error, since this function returns a `Result<bool, Error>

Reply:
Odd. What was the error?

Reply:
If I use the `Error::cli` function here I end up with the following error:

```
expected type `std::result::Result<bool, _>`
found type `std::result::Result<(), _>`
```

This is happening because that function wraps the error up into a result, which is fine for functions that also return a `Result<(), Error>`, but means that it won't work for functions that return something else in the `Ok` case :sad`

```rust
pub fn cli(message: &str, stderr: Cow<str>) -> Result<(), Self> {
```